### PR TITLE
Bump version of golangci-lint to v1.51.0

### PR DESCRIPTION
**Pull request by siredmar**
Fixes https://github.com/golangci/golangci-lint/issues/3565

### CODE REVIEW

**Comment by siredmar** [comment]
@turkenh would you mind reviewing this PR?

File: makelib/golang.mk [hunk 1]
```diff
@@ -98,7 +98,7 @@ endif
 
 # We use a consistent version of golangci-lint to ensure everyone gets the same
 # linters.
-GOLANGCILINT_VERSION ?= 1.50.1
+GOLANGCILINT_VERSION ?= 1.51.0
```
Comment (hasheddan) [inline makelib/golang.mk:101]: Thanks @siredmar! If you are encountering this in a repo you can always override in your parent `Makefile` by explicitly setting `GOLANGCILINT_VERSION` as well :) Also, any reason not to go ahead and bump to latest (`1.52.2`)?

Comment (siredmar) [reply makelib/golang.mk:101]: I guess the reason for this was the lined version in the issue mentioned above. I didn't test go1.20 with 1.52.2. If you like, you can test it as i don't have the time to test this right now.

**Review by hasheddan — APPROVED** [review]
This will get us past the linked bug — we can bump to a newer version as needed. Thanks @siredmar!

**Pull request merged**